core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -381,7 +381,39 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments,
                             entityName: String,
                             newName: String,
                             parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] = {
+    // Check if the newName is the same as the old name
+    if (newName == entityName) {
+      throw new RawlsExceptionWithErrorReport(
+        errorReport = ErrorReport(StatusCodes.BadRequest, s"New name $newName is the same as the old name $entityName")
+      )
+    }
+    // Validate the newName
+    EntityUtils.validateEntityName(newName)
+    // Perform the rename in a transaction
+    val renameFuture = repository.dataSource.inTransaction { _ =>
+      for {
+        // Check if the newName already exists
+        newNameExists <- repository.queries.getEntity(workspaceId, entityType, newName)
```

Comment:
nit: use `existsAll()` instead of `getEntity()`, since you don't care about the _content_ of the entity, you only care if it exists. This only really matters for large entities like sample sets; you don't have to return the entire contents of the entity in the db result.